go/vt/vtgate/planbuilder/operators/subquery.go

```diff
@@ -253,6 +253,20 @@ func (sq *SubQuery) settleFilter(ctx *plancontext.PlanningContext, outer Operato
 	}
 	post := func(cursor *sqlparser.CopyOnWriteCursor) {
 		node := cursor.Node()
+		// For IN and NOT IN type filters, we have to add a Expression that checks if we got any rows back or not
+		// for correctness. That expression should be ANDed with the expression that has the IN/NOT IN comparison.
+		if compExpr, isCompExpr := node.(*sqlparser.ComparisonExpr); sq.FilterType.NeedsListArg() && isCompExpr {
+			if listArg, isListArg := compExpr.Right.(sqlparser.ListArg); isListArg && listArg.String() == sq.ArgName {
```

Comment:
Should we not check for either the right or left hand side of the outer's comparison? The query we are trying to support contains the subquery comparison on the RHS of the `OR` but what if the query was the following:
```sql
# new query
select count(*) from user where user.id in (select id from unsharded_a where colb = 2) or user.id = 2

# instead of
select count(*) from user where user.id = 2 or user.id in (select id from unsharded_a where colb = 2)
```

This comment applies to both this code and the code later in the same function where we construct the list of predicates, line `298`.

Reply:
The comparision operation in question is `user.id IN (subquery)`, and the right side in this is the subquery. And the subquery can't be on the left side as in `(subquery) IN user.id`.

Reply:
Makes sense! Thank you.